### ISSUE TITLE
Updates to the Concept Scheme for ceterms:CreditUnit

### ISSUE THREAD

**Issue by cwd-mparsons** [CredentialEngine/Schema-Development#590]
The concept scheme for creditUnitType will have to be updated to include, at least the suggestions from credit hours

- semester hours,
- quarter hours
- clock hours
- hours of participation
- contact hours

**Comment by jkitchensSIUC** [comment]
@jeff-grann this is a secondary and postsecondary  issue.  Can you look into this and determine if these should be included with the existing concept scheme https://credreg.net/ctdl/terms#CreditUnit.

**Comment by siuc-nate** [comment]
Per our 4-2-2019 meeting:
If a value doesn't match one of the controlled values, use the description property inside of QuantitativeValue instead.

**Comment by siuc-nate** [comment]
Per our 4-9-2019 meeting:
@jeff-grann We need you to have some "starter" definitions for each of these terms.

**Comment by cwd-mparsons** [comment]
@jeff-grann 
I did some searching for possible definitions

Hours of Participation
One CEU equals ten contact hours of participation in organized continuing education classes and/or training conducted by a qualified instructor.

Quarter Hours
Quarter credit hours are granted in place of semester credit hours by schools that utilize a quartered calendar system.

Semester Hours
A unit of academic credit representing an hour of class (such as lecture class) or three hours of laboratory work each week for an academic semester.

Clock Hours
Clock hours are the total number of actual hours per week a student spends attending class or other instructional activities that count toward completing a program of study.

Contact Hours
A standard one hour (at least 50 minutes) classroom period.

**Comment by siuc-nate** [comment]
@stuartasutton @jkitchensSIUC We're waiting to hear back from the other group before we do anything in the schema for this one, correct?

**Comment by siuc-nate** [comment]
Based on feedback ( https://docs.google.com/document/d/1i3CnKXC3oOF2Iu_d3ZOmuLNQjT91n8I1OWj3tltlUEU/edit ), the following have been proposed:

**URI:** creditUnit:CreditHour
**Label:** Credit Hour
**Definition:** A quantity of student learning used for awarding U.S. Federal student aid.
**Usage Note:** According to U.S. Federal regulation 34 CFR § 600.2 Definitions:
Credit hour: Except as provided in 34 CFR 668.8(k) and (l), a credit hour is an amount of work represented in intended learning outcomes and verified by evidence of student achievement that is an institutionally established equivalency that reasonably approximates not less than -
One hour of classroom or direct faculty instruction and a minimum of two hours of out of class student work each week for approximately fifteen weeks for one semester or trimester hour of credit, or ten to twelve weeks for one quarter hour of credit, or the equivalent amount of work over a different amount of time; or
At least an equivalent amount of work as required in paragraph (1) of this definition for other academic activities as established by the institution including laboratory work, internships, practica, studio work, and other academic work leading to the award of credit hours.
https://www.ecfr.gov/cgi-bin/text-idx?rgn=div8&node=34:3.1.3.1.1.1.23.2
Program Integrity Questions and Answers - Credit Hour
https://www2.ed.gov/policy/highered/reg/hearulemaking/2009/credit.html#credit


**URI:** creditUnit:SemesterHour
**Label:** Semester Hour
**Definition:** A unit of credit representing an hour of class (such as a lecture class) or three hours of laboratory work each week for an academic semester.
**Comment:** The credit hour concept for institutions using a semester-based academic calendar.
Subconcept Of: creditUnit:CreditHour


**URI:** creditUnit:QuarterHour
**Label:** Quarter Hour
**Definition:** A unit of credit representing an hour of class (such as a lecture class) or three hours of laboratory work each week for an academic quarter.
**Comment:** The credit hour concept for institutions using a quarter-based academic calendar.
Subconcept Of: creditUnit:CreditHour

**Comment by siuc-nate** [comment]
One of our initial steps was to rework the Credit Hour proposal to be simpler and non-US-centric:

**URI:** creditUnit:CreditHour
**Label:** Credit Hour
**Definition:** A time-based quantity of student learning.
**Usage Note:** For usage in the U.S., see: https://www.ecfr.gov/cgi-bin/text-idx?rgn=div8&node=34:3.1.3.1.1.1.23.2 and https://www2.ed.gov/policy/highered/reg/hearulemaking/2009/credit.html#credit

**Comment by mparsons-ce** [comment]
@jeannekitchens @stuartasutton @siuc-nate 
Is this issue ready to be closed/added to the next CTDL release?

**Comment by siuc-nate** [comment]
1. Is the assertion of "one hour of lecture or three hours of laboratory work" sufficiently universal that we won't need to make more concepts later (e.g. "our semester hour includes one hour of lecture or one hour of lab" or "our credit hour consists of one hour of each")?

2. We should genericize the usage note for ClockHour so it's not so verbose or US-specific

3. We should examine any other terms from Mike's findings #590

**Comment by stuartasutton** [comment]
No it is not universal. The ratio of face-to-face instruction to supervised activity and/or unsupervised external hours (i.e., homework) are not standardized. So, I might have a course worth 4 semester hours that means 3 hours of lecture and 15 hours of outside study  (i.e. homework) per week...or 2 hours of lecture, 3 hours of lab and 10 hours of outside study per week. In the U.S., the semester and quarter credits represent some identified ratio of face-to-face instruction (e.g., lecture, seminar etc.), and/or supervised activity (e.g., lab) and/or unsupervised activity (e.g., homework). 

And, graduate credit hours have higher, more strenuous hours than undergraduate. E.g., at University of Washington, 3 credit hours of undergraduate study might have an expectation of 3 lecture hours and 6 hours of unsupervised activity (e.g., homework) while graduate student might have an expectation of 3 lecture hours and 12 hours of unsupervised activity. And, a class might be a 3 credit hour quarter course with a mix of both  graduate and undergraduate students earning the same number of  credit hours with different expectation ratios on hours of actual work.

So, definitions should not hardwire ratios or the exact aspects on which those ratios are based.

**Comment by mparsons-ce** [comment]
The below image shows the counts (from credential finder) for creditUnitTypeDescriptions that are less than 50 characters in length for learning opportunities, condition profiles, and assessments:

![image](https://user-images.githubusercontent.com/58825866/76907949-89c3e300-6875-11ea-9445-07c846aaddba.png)

**Comment by philbarker** [comment]
The terms in currently in the [CreditUnit](https://credreg.net/ctdl/terms/CreditUnit#CreditUnit) concept scheme seem to relate to the educational level of the credit; the entries Mike is finding in the description relate to the extent of the work involved to earn one. I think **these are two orthogonal concepts**.

For example, in Scotland we have the SCQF credits represent a notional 10 hours of learning time and come in 12 levels (early years & special needs education through to PhD). Typically to be awarded an Honours Degree you need "at least 480 credits of which a minimum of 180 is at levels 9 and 10, including at least 90 at level 10". 

So to know the value of your credits in Scotland you need to know how your credit unit equates to SCQF Credits you have **and** at what level.

For comparison in the rest of Europe, there are 8 levels and the credits represent ~20 hours study time, so some conversion is required.

**Comment by siuc-nate** [comment]
In my opinion, if they need their own vocabulary, they will also need their own property to point at said vocabulary within schema:QuantitativeValue.  Otherwise, we can put them in the same vocabulary and ensure that schema:unitText is multi-select.

**Comment by philbarker** [comment]
Example of credit scheme in use for credentials: [Scottish Qualifications ready reckoner](https://www.sqa.org.uk/sqa/files_ccc/B63338_SQA_A6_ready%20reckoner.pdf) (pdf) p2 National Qualifications. These are certificates awarded after course & assessment in a single subject at school. Normally Nationals 1-5 are taken at age 15-16, Highers at 16-17 and Advanced Highers at 17-18. 

See how the level and number of credit points varies, e.g.:
National 1: 6 points at level 1
National 5: 24 points at level 5
Higher: 24 points at level 6
Advanced Higher: 32 points at level 7

"Levels (1-12) show how complex the learning of a qualification is. 

Credit points indicate the volume of learning required to achieve a qualification.
One SCQF credit point represents 10 notional learning hours."

The time take varies between qualifications, students at high school typically take 6-8 nationals over two years; 4 or 5 Highers in one year; 3 Advanced Highers in one year.

**Comment by philbarker** [comment]
Example of credit scheme for credentials & course/programs [SQA Higher National Qualifications](https://www.sqa.org.uk/sqa/411.html) (NB, these are not the same as 'highers' or 'nationals' in previous example).
Higher National Diploma requires 240 credit points at levels 7 & 8 (2yr course)
Higher National Certificate requires 96 credit points at level 7 (1yr course)

These are "group awards" ie are awarded for after program of study (which here we call a course) comprising several courses (here called Higher National Units) and assessments. HN Units are typically 8 or 16 credit points.

For example, [HND/C in Cyber Security](https://www.sqa.org.uk/sqa/84394.html) ([full spec in pdf](https://www.sqa.org.uk/sqa/files_ccc/GP8316.pdf)). From the full spec:

> This group award is made up of 30 SQA unit credits. 17 credits (136 SCQF credit points) are mandatory, including two graded units of 3 SQA credits (24 SCQF credit points) at SCQF
levels 7 and 8, and must be taken by all learners, and 13 credits are optional (104 SCQF
credit points), which are selected from the list of optional units.

[Unit credits ~ HNC Units, but some units offer double credits--they're twice as long. This is largely redundant as it is also reflected in the credit points.]

**Comment by philbarker** [comment]
final example, at [University level](https://www.gla.ac.uk/myglasgow/senateoffice/qea/progdesignapproval/scqf/detailsofthequalificationsateachlevelofhighereducation-appendix2/), e.g. 
The Scottish Bachelors Degree with Honours [equiv to US 4 year degree]
SCQF Level: 	10 (SHE level H)
Credit Points: 	At least 480 credits of which a minimum of 180 is at levels 9 and 10, including at least 90 at level 10.

**Comment by philbarker** [comment]
Additional information: there was a CEN Workshop Agreement for an [Educational Credit Information Model](https://webstore.ansi.org/Standards/DS/DSCWA160772010) (2010, now lapsed). It was a single class `ecim:Credit` with three properties `ecim:scheme`, `ecim:level`, `ecim:value` all of which were Literals, AFAIK.

**Comment by siuc-nate** [comment]
That's a lot of useful context, thank you.  What would some JSON examples look like for the QuantitativeValue class to carry this data (making up whatever properties are necessary to carry the values)?

**Comment by philbarker** [comment]
Here are a couple of options, first making do with QuantitativeValue, second moving farther from that in order to provide more precision. (NB: URIs for scqf terms are made up)

```
{
  "@type": "ceterms:SQANational5",
  "ceterms:name": {
      "en-US": "National 5 Certificate"
    },
  "ceterms:requires": {
    "@type": "ceterms:ConditionProfile",
    "ceterms:creditValue": {
      "@type": "schema:QuantitativeValue",
      "schema:unitText": "SCQF Credit Points",
      "schema:value": 24,
      "schema:educationalLevel": {
        "@type": "schema:DefinedTerm",
        "@id": "http://scqf.org.uk/the-framework/#level5",
        "schema:name":"5",
        "inDefinedTermSet": "SCQF Levels"
      }
    }
  }
}


{
  "@type": "ceterms:SQANational5",
  "ceterms:name": {
      "en-US": "National 5 Certificate"
    },
  "ceterms:requires": {
    "@type": "ceterms:ConditionProfile",
    "ceterms:creditValue": {
      "@type": "ceterms:Credit",
      "ceterms:framework": "http://scqf.org.uk/the-framework/#credits",
      "ceterms:frameworkName": "SCQF Credit Points",
      "schema:value": 24,
      "schema:educationalLevel": {
        "@type": "schema:DefinedTerm",
        "@id": "http://scqf.org.uk/the-framework/#level5",
        "schema:name":"5",
        "inDefinedTermSet": "SCQF Levels"
      }
    }
  }
}
```

And here's start at diagramming, which doesn't quite align with either option in JSON (sorry, ideas still in flux, and I'm drifting between schema.org-like and ctld-like approaches) but may help.
![credits](https://user-images.githubusercontent.com/658047/78266237-677cc700-74fd-11ea-94a3-44dd03d78f04.png)

**Comment by siuc-nate** [comment]
I noticed that the schema.org definition for https://schema.org/numberOfCredits now allows for either an integer or a https://schema.org/StructuredValue - is that new?  It seems like that would cover our needs for describing the credits in more detail.

**Comment by philbarker** [comment]
@siuc-nate yeah schema has allowed the use of StructuredValue since numberOfCredits was added, and that may cover what we want, but imperfectly and only partially, I think, because there's no clear place to put educational level, and no unambiguous way to identify the scheme being used. (Plus there is the other issue that even though schema allows it, Google job training pathways doesn't)

**Comment by siuc-nate** [comment]
I _think_ the intent would be to use the schema:identifier property to identify both the type and level (as two separate values for that property) - it's not as concrete as schema:educationalLevel, but it does allow flexibility for other kinds of credit.

As for Google, we may have an easier time convincing them to use a range that's already prescribed versus getting them to adopt a CTDL class.

If I were to write a converter to output a schema.org record (similar to the EOCred converter) for a learning opportunity, I would (if only due to lack of having any other choice) use StructuredValue to carry the credit information.

**Comment by philbarker** [comment]
Not sure I understand the case for using `identifier` for type and level. These are more about classification than identification. I think QuantitativeValue as described above is a better match, especially where no information about level is required (and you could get around that in some cases by atteching the level to the thing being described, e.g. Course when decsribing credits awaded).

To me adding CreditValue to schema.org as a subtype of StructuredValue is the way to go as far as Google pathways is concerned (which is why I suggested they include StructuredValue in the range of numberOfCredits).

We can probably do this through the EOCred W3C Community Group, but useful to sort out usecases first and make sure that those align with CTDL use.

**Comment by siuc-nate** [comment]
If I'm reading it right, `QuantitativeValue` is a subclass(?) of `StructuredValue`.  Does that mean it's acceptable (in terms of how schema.org intends things to be used) to use a `QuantitativeValue` anywhere something expects a `StructuredValue`? Or would `QuantitativeValue` need to be explicitly listed as a possible value for properties like `numberOfCredits`?

Either way, it looks like QuantitativeValue has an `identifier` property as well.  To answer your question, my intent there would be to use it to identify the type of credit and identify any other relevant aspects of credit (such as educational level) as properties of the QuantiativeValue class that carries the numeric value for the credits themselves - but I'm not sure (from the definition) whether that would be interpreted as an identifier for the credits or an identifier for the QuantitativeValue object itself.

**Comment by siuc-nate** [comment]
Okay, I tried to work out an example, and I probably have to backpedal a little bit, because without some useful property to reference an existing term in a vocabulary somewhere that can provide context for what "SCQF Credits" and "Level 5" actually mean, this does fall a little flat:
```
{
  "@type": "ceterms:Certificate",
  "ceterms:requires": [
    {
      "@type": "ceterms:ConditionProfile",
      "ceterms:creditValue": [
        {
          "@type": "schema:QuantitativeValue",
          "schema:value": 24,
          "schema:description": { "en": "24 SCQF Level 5 Credits" },
          "schema:identifier" [
            {
              "@type": "schema:PropertyValue",
              "schema:unitText": { "en": "SCQF Credits" }
            },
            {
              "@type": "schema:PropertyValue",
              "schema:unitText": { "en": "Level 5" }
            }
          ]
        }
      ]
    }
  ]
}
```

**Comment by philbarker** [comment]
> Does that mean it's acceptable (in terms of how schema.org intends things to be used) to use a QuantitativeValue anywhere something expects a StructuredValue?

Yes, because a member of a sub class is a member of the super class. All QuantitativeValues are StructuredValues. This is worth remembering where the range is a high-level type such as Thing or CreativeWork.

> QuantitativeValue has an identifier property as well. To answer your question, my intent there would be to use it to identify the type of credit ...

That doesn't sound right. You need an identifier for the QuantitativeValue itself, not an idenitifier for the type of QuanititiveValue. That is to say, an identifier is not a classifier. An identifier must identify a unique entity. In this case it would be like giving "6kg" and identifier.

**Comment by philbarker** [comment]
Second example is better, but "Level 5" is not a unit in which credits are measured. 

Best you could do is `"schema:unitText" : "SCQF Level 5 Credits"` but that (a) makes an arbitrary change to how SCQF deals with credits in order to fit the schema, and (b) needs out of band info to interpret what SCQF credits are.  You might as well just stop with `"schema:description": "24 SCQF Level 5 Credits" `.

**Comment by siuc-nate** [comment]
>Yes, because a member of a sub class is a member of the super class. All QuantitativeValues are StructuredValues. This is worth remembering where the range is a high-level type such as Thing or CreativeWork.

Good to know - I wasn't sure if that was the case or not since the notion of inheritance doesn't always work the same way across schemas/languages/systems/etc.  Can something in schema.org be a subclass of more than one parent class (regardless of where it may get its properties from)? 

>That doesn't sound right. You need an identifier for the QuantitativeValue itself, not an idenitifier for the type of QuanititiveValue.

Yeah, I see what you mean.

So, it would appear we do need to, as you suggest, request a new class in schema.org to describe educational credits specifically.  Are there other notions besides `educationalLevel` that such a class should cover?  I don't know enough about the world of academic credit to suggest an example.

Regardless, I think such properties would be most effective if they leveraged skos:Concepts or some similar means of (externally) defining and referencing things like levels, since "Level 5" can mean so many different things in different situations.

Is schema:DefinedTerm meant to replace skos:Concept? It seems the two cover a lot of the same ground.

**Comment by siuc-nate** [comment]
>Best you could do is "schema:unitText" : "SCQF Level 5 Credits" but that (a) makes an arbitrary change to how SCQF deals with credits in order to fit the schema, and (b) needs out of band info to interpret what SCQF credits are.

I think part of this will depend on how specific those "levels" are to SCQF credits.  If they're generic enough that they could be applied to other kinds of credits, then these would make sense as two distinct types of notion - one for the credit type and one for the level, that could be mixed and matched as needed.  However, if the levels are closely tied to the SCQF credits, they may make sense as part of the same concept scheme, e.g.:

- Continuing Education Unit
- Degree Credit
- Other types of Credit...
- SCQF Credit
  - SCQF Level 1
  - SCQF Level 2
  - SCQF Level 3
  - etc...

In other words, we would need to define levels to work independently of SCQF credits in order for them to be useful as a standalone concept scheme, I think.  Again, I don't know enough about how that all works to guess at whether they can stand on their own or not.

**Comment by philbarker** [comment]
>  Can something in schema.org be a subclass of more than one parent class (regardless of where it may get its properties from)?

Yes, in which case it inherits properties from both parent classes. An example is https://schema.org/LocalBusiness which is both a Organization and a Place (as in usage "meet me at McDonald's").  You can also declare something to be more than one type of thing when describing it, e.g. a Product and a CreativeWork, which lets you use properties from both types. 

> Regardless, I think such properties would be most effective if they leveraged skos:Concepts or some similar means of (externally) defining and referencing things like levels, since "Level 5" can mean so many different things in different situations....
> Is schema:DefinedTerm meant to replace skos:Concept? 

Agreed, and yes it's similar (a couple of the examples I gave above use it for this).

> we would need to define levels to work independently of SCQF credits in order for them to be useful as a standalone concept scheme

Levels are indeed independent of credits, and are applied to things other than credits

**Comment by philbarker** [comment]
Folks, a straw man proposal for some new terms to allow the description of level and unit of Credits:

**New Class**
> **URI:** ceterms:CreditValue
> **Label:** Credit Value
> **Definition:** A value for the amount of credit required in a Condition Profile or awarded for successful completion of a Learning Opportunity or Assessment.
> **Type of Term:** rdfs:Class
> **Subclass Of:** ceterms:QuantitativeValue
> **Properties:** ceterms:creditUnitType ceterms:creditLevelType ceterms:description qdata:percentage schema:description schema:maxValue schema:minValue schema:unitText schema:value
> **In Range Of:** ceterms:creditValue

**Updated Properties:**
**Add:**
> **Subject:** ceterms:creditValue
> **Predicate:** schema:rangeIncludes
> **Object:** ceterms:CreditValue

**New properties**

> **URI:** ceterms:creditUnitType
> **Label:** Credit Unit Type
> **Definition:** The type of credit associated with the credit awarded or required.
> **Domain:** ceterms:CreditValue
> **Range:** schema:DefinedTerm 

> **URI:** ceterms:creditLevelType
> **Label:** Credit Level Type
> **Definition:** The level of credit associated with the credit awarded or required.
> **Domain:** ceterms:CreditValue
> **Range:** schema:DefinedTerm

**Samples**
1. this is based on the PESC example Nate showed in the May 5, 2020 team meeting [notes](https://docs.google.com/document/d/1nCUyx4Tr8wTgOMeBDddZwjtCjSUzsovohmEyd0b57GY/edit#heading=h.abq2ketbbpzw)

```
{
  "@context": [
    { "schema": "http://schema.org/"},
    { "ceterms": "https://purl.org/ctdl/terms/"}
  ],
  "@graph": {
    "@type": "ceterms:LearningOpportunityProfile",
    "ceterms:creditValue":  {
      "@type": "ceterms:CreditValue",
      "schema:value": 4.0,
      "ceterms:creditUnitType": {
        "@type": "schema:DefinedTerm",
        "@id": "https://ceds.ed.gov/element/000057#Semester",
        "schema:name": "Semester",
        "schema:inDefinedTermSet": {
          "@type": "schema:DefinedTerm",
          "schema:name": "CEDS"
        }
      },
      "ceterms:creditLevelType": {
        "@type": "schema:DefinedTerm",
        "@id": "https://ceds.ed.gov/element/001312#LowerDivision",
        "schema:name": "Lower division",
        "schema:inDefinedTermSet": {
          "@type": "schema:DefinedTerm",
          "schema:name": "CEDS"
        }
      }
    }
  }
}
```
Read as: A learning opportunity offers a credit value of 4.0 [Semester](https://ceds.ed.gov/element/000057#Semester) units as defined by CEDS at a credit level [Lower division](https://ceds.ed.gov/element/001312#LowerDivision) as defined by CEDS.


2. requirements for SQA National 5 Certificate (see [above](https://github.com/CredentialEngine/Schema-Development/issues/590#issuecomment-607921197))
```
{
  "@context": [
    { "schema": "http://schema.org/"},
    { "ceterms": "https://purl.org/ctdl/terms/"}
  ],
  "@graph": {
    "@type": "ceterms:Certificate",
    "ceterms:name": {
        "en-US": "National 5 Certificate"
    },
    "ceterms:requires": {
      "@type": "ceterms:ConditionProfile",
      "ceterms:creditValue": {
        "@type": "ceterms:CreditValue",
        "schema:value": 24,
        "ceterms:creditUnitType": {
          "@type": "schema:DefinedTerm",
          "schema:name": "Credit Points",
          "schema:inDefinedTermSet": {
            "@type": "schema:DefinedTerm",
            "schema:name": "Scottish Credit and Qualifications Framework",
            "url": "https://scqf.org.uk/the-framework/about-the-framework/"
          }
        },
        "ceterms:creditLevelType": {
          "@type": "schema:DefinedTerm",
          "schema:name": "Level 5",
          "schema:inDefinedTermSet": {
            "@type": "schema:DefinedTerm",
            "schema:name": "Scottish Credit and Qualifications Framework",
            "url": "https://scqf.org.uk/the-framework/about-the-framework/"
          }
        }        
      }
    }  
  }
}
```
Read as: A National 5 Certificate requires a credit value of 24 Credit Points as defined by the [Scottish Credit and Qualifications Framework](https://scqf.org.uk/the-framework/about-the-framework/), at a credit level of Level 5 as defined by the [Scottish Credit and Qualifications Framework](https://scqf.org.uk/the-framework/about-the-framework/)

**Comment by siuc-nate** [comment]
Moving this to the June milestone, as the solution is not yet determined.

**Comment by siuc-nate** [comment]
Per our 8/19/2020 meeting:

Determine how best to document credit like the following:
>In the upper-division baccalaureate degree category, 2 semester hours in electrical engineering, data communications and networking, telecommunications, or management information systems (11/90).

Proposal Examples:
- A combination of Unit Type, Unit Level, and potentially Unit Period
```
{
  "Value": 2.0,
  "UnitText": [        
    {
      "TargetNode": "creditType:DegreeCredit" //This is redundant with Bachelors Degree Level below
    }
  ],
  "UnitLevel": [
    {
      "TargetNode": "audLevel:UpperDivision"
    },
    {
      "TargetNode": "audLevel:BachelorsDegreeLevel"
    }
  ],
  "UnitPeriod": [
    {
      "TargetNode": "period:Semester"
    },
    {
      "TargetNode": "period:Hour"
    }
  ],
  "Description": "In the upper-division baccalaureate degree category, 2 semester hours in electrical engineering, data communications and networking, telecommunications, or management information systems (11/90)."
}

```

- A class of concept schemes that embodies the above but is left up to the user to define specific terms
```
{
  "Value": 2.0,
  "UnitText": [
    {
      "TargetNode": "https://acenet.edu/credittype/upperdivision"
    },
    {
      "TargetNode": "https://acenet.edu/credittype/baccalaureate"
    },
    {
      "TargetNode": "https://acenet.edu/credittype/semesterhour"
    }
    
  ]
  "Description": "In the upper-division baccalaureate degree category, 2 semester hours in electrical engineering, data communications and networking, telecommunications, or management information systems (11/90)."
}
```

- A combination of our terms and user-defined terms:
```
{
  "Value": 2.0,
  "UnitText": [
    {
      "TargetNode": "https://acenet.edu/credittype/upperdivision"
    },
    {
      "TargetNode": "audLevel:BachelorsDegree"
    },
    {
      "TargetNode": "https://acenet.edu/credittype/semesterhour"
    }
  ]
  "Description": "In the upper-division baccalaureate degree category, 2 semester hours in electrical engineering, data communications and networking, telecommunications, or management information systems (11/90)."
}
```

**Comment by siuc-nate** [comment]
Additional example: https://www.ece.org/ECE-FIles/Sample-Reports/Sample_CBC_Report.pdf
"Three years of undergraduate study with a grade average of 2.72"

**Comment by siuc-nate** [comment]
Additional example: http://interstatepassport.wiche.edu/documents/IP_Overview.pdf
"63 Passport Learning Outcomes" (competencies) or "1 Passport" or some other measure

**Comment by jeannekitchens** [comment]
@siuc-nate please use a Google doc and just as we showed updates to financial assistance use the same format as follows:

1. show in black text all existing relevant concepts with their definitions and comments.
2. show all proposed changes using red text for new terms and definition and comment revisions interspersed with this GitHub issue.
3. use the Transfer Value Google Sheet to include the upper and lower division.
4. show the structure as a heirarchy if it's applicable.
5. link to this issue and we'll review to create a final proposal to review and move this forward with Transfer Value.

**Comment by siuc-nate** [comment]
I created a document here:
https://docs.google.com/document/d/1LC6_GbKdYWYwLoGtYAUahRxWdSJ2By815Br_-iQ1mGw/edit#

This document assumes an approach where:
- Regardless of whether we continue to use schema:QuantitativeValue or some new class (e.g. ceterms:CreditValue), there will be at least two properties that reference the relevant concept schemes:
  - schema:unitText (or we could un-deprecate [ceterms:creditUnitType](https://credreg.net/ctdl/terms/creditUnitType)), which points to values in the Credit Unit concept scheme
  - ceterms:audienceLevelType (or a new property like ceterms:creditLevelType, which would be more appropriate if we use schema:QuantitativeValue but less necessary if we create a ceterms:CreditValue class), which points to values in the Audience Level concept scheme
  - Potentially some third property to describe more specific amounts of time to cover credit that spans specific periods (e.g. the example above of "Three years of undergraduate study")

Examples:

```
{
  "ceterms:creditValue": [
    {
      "@type": "schema:QuantitativeValue",
      "schema:description": { "en": "Three years of undergraduate study with a grade average of 2.72" },
      "ceterms:creditLevelType": [
        {
          "@type": "ceterms:CredentialAlignmentObject",
          "ceterms:targetNode": "audLevel:Undergraduate"
        }
      ],
      "schema:value": 3.0,
      "schema:unitText": [
        {
          "@type": "ceterms:CredentialAlignmentObject",
          "ceterms:targetNodeName": "Years"
        }
      ]
    }
  ]
}
```

```
{
  "ceterms:creditValue": [
    {
      "@type": "ceterms:CreditValue",
      "ceterms:description": "63 Passport Learning Outcomes",
      "ceterms:creditUnitType": [
        {
          "@type": "ceterms:CredentialAlignmentObject",
          "ceterms:targetNode": "creditUnit:Competency"
        }
      ],
      "schema:value": 63.0
    }
  ]
}
```

```
{
  "ceterms:creditValue": [
    {
      "@type": "ceterms:CreditValue",
      "ceterms:description": "24 Credit Points at Level 5 as described by the Scottish Credit and Qualifications Framework.",
      "ceterms:creditUnitType": [
        {
          "@type": "ceterms:CredentialAlignmentObject",
          "ceterms:targetNodeName": { "en": "Credit Points" },
          "ceterms:framework": "https://scqf.org.uk/the-framework/about-the-framework/",
          "ceterms:frameworkName": { "en": "Scottish Credit and Qualifications Framework" }
        }
      ],
      "ceterms:creditLevelType": [
        {
          "@type": "ceterms:CredentialAlignmentObject",
          "ceterms:targetNodeName": { "en": "Level 5" },
          "ceterms:framework": "https://scqf.org.uk/the-framework/about-the-framework/",
          "ceterms:frameworkName": { "en": "Scottish Credit and Qualifications Framework" }
        }
      ],
      "schema:value": 24
    }
  ]
}
```

```
{
  "ceterms:creditValue": [
    {
      "@type": "ceterms:CreditValue",
      "ceterms:description": "In the upper-division baccalaureate degree category, 2 semester hours in electrical engineering, data communications and networking, telecommunications, or management information systems (11/90).",
      "ceterms:creditUnitType": [
        {
          "@type": "ceterms:CredentialAlignmentObject",
          "ceterms:targetNode": "creditUnit:DegreeCredit"
        },
        {
          "@type": "ceterms:CredentialAlignmentObject",
          "ceterms:targetNode": "creditUnit:SemesterHour"
        }
      ],
      "ceterms:creditLevelType": [
        {
          "@type": "ceterms:CredentialAlignmentObject",
          "ceterms:targetNode": "audLevel:UpperDivision"
        }
      ],
      "schema:value": 2
    }
  ]
}
```

**Comment by siuc-nate** [comment]
Based on @philbarker's proposal:

**New Class**
> **URI:** ceterms:CreditValue
> **Label:** Credit Value
> **Definition:** A value and description of credit awarded for, required by, or otherwise related to the resource.
> **Type of Term:** rdfs:Class
> **Subclass Of:** ceterms:QuantitativeValue
> **Properties:** ceterms:creditUnitType ceterms:creditLevelType qdata:percentage schema:description schema:maxValue schema:minValue schema:unitText schema:value
> **In Range Of:** ceterms:creditValue

**Updated Properties:**
**Remove:**
> **Subject:** ceterms:creditValue
> **Predicate:** schema:rangeIncludes
> **Object:** schema:QuantitativeValue

**Add:**
> **Subject:** ceterms:creditValue
> **Predicate:** schema:rangeIncludes
> **Object:** ceterms:CreditValue

**New properties**

> **URI:** ceterms:creditUnitType
> **Label:** Credit Unit Type
> **Definition:** The type of credit associated with the credit awarded or required.
> **Domain:** ceterms:CreditValue
> **Range:** skos:Concept 

> **URI:** ceterms:creditLevelType
> **Label:** Credit Level Type
> **Definition:** The level of credit associated with the credit awarded or required.
> **Domain:** ceterms:CreditValue
> **Range:** skos:Concept

**Comment by siuc-nate** [comment]
@jeannekitchens @stuartasutton @philbarker @mparsons-ce 
I have updated the post above with the final proposal for the class, properties, and concept updates.

**Comment by jeannekitchens** [comment]
Thank you @siuc-nate.  Could you also link the updated domain model for Transfer Value Profile using CreditUnit?

**Comment by siuc-nate** [comment]
This should capture it:
https://app.diagrams.net/#G1Yg0qKQAQnK8Qev2rWYKQJfO9NvOQUM7_

![image](https://user-images.githubusercontent.com/21346519/94060515-b2d7aa00-fda9-11ea-99ef-f81f448c8504.png)

**Comment by siuc-nate** [comment]
These changes have been made in pending CTDL and noted in the history tracking.
